src/app/treasury/components/AddressLink.tsx

```diff
@@ -0,0 +1,65 @@
+import { CopySvg } from '@/components/CopySvg'
+import { ArrowUpRightLightIcon } from '@/components/Icons'
+import { Span } from '@/components/TypographyNew'
+import { EXPLORER_URL } from '@/lib/constants'
+import { cn, shortAddress } from '@/lib/utils'
+import { useEffect, useState } from 'react'
+import { Address } from 'viem'
+
+const BASE_CLASSNAME = 'hover:text-text-60 transition-colors duration-100'
+
+export function AddressLink({ address, className }: { address: Address; className?: string }) {
```

Comment:
Could we expand on the documentation of this component?

Reply:
done